### ISSUE TITLE
Serverless Error:  An error occurred: UserLambdaPermissionApiGateway - The final policy size (20581) is bigger than the limit (20480). (Service: AWSLambda; Status Code: 400; Error Code: PolicyLengthExceededException;.

### ISSUE THREAD

**Issue by akshay-scalex** [serverless/serverless#7411]
# Bug Report

## Description
I am using serverless framework for couple of years but I never faced this issue before this.
I have one service deployed on AWS Lambda using serverless framework before a month, there was a bug in that service. I fixed the bug and tried redeploying it. It gave me an error:
`Serverless Error ---------------------------------------    An error occurred: UserLambdaPermissionApiGateway - The final policy size (20581) is bigger than the limit (20480). (Service: AWSLambda; Status Code: 400; Error Code: PolicyLengthExceededException; Request ID: 85b45e19-8835-4d0d-923b-04234e8378fa).`

1. What did you do?
Tried to reploy existing micro-service without any change in serverless.yml file. Just updated one database query which is talking to a postgresql over connection string.

2.  What happened?
After updating the two lines in code, I tried deploying it again. It gave me an error:
`Serverless Error ---------------------------------------    An error occurred: UserLambdaPermissionApiGateway - The final policy size (20581) is bigger than the limit (20480). (Service: AWSLambda; Status Code: 400; Error Code: PolicyLengthExceededException; Request ID: 85b45e19-8835-4d0d-923b-04234e8378fa).`

3. What should've happened?
It should have deployed the micro-service again without any error

4. What's the content of your `serverless.yml` file?
-----------------------------------------------------------
```
service: stack-name


provider:
  name: aws
  runtime: nodejs10.x
  timeout: 60
  stage: ${opt:stage}
  environment: ${file(env.yml):${self:custom.stage}}
  region: ap-south-1
  deploymentBucket:
    name: ${self:provider.environment.DEPLOYMENT_BUCKET}
  iamRoleStatements:
    - Effect: Allow
      Action:
        - S3:*
      Resource: "*"

functions:
  fun1:
    handler: src/f1.function1
    events:
      - http:
          method: post
          path: service/f1
          authorizer: # An AWS API Gateway custom authorizer function
            arn:  arn:aws:lambda:${opt:region, self:provider.region}:${self:provider.environment.ACCOUNT_ID}:function:${self:provider.environment.AUTHORIZER_FUNCTION} # Can be used instead of name to reference a function outside of service
            resultTtlInSeconds: 0
            identitySource: method.request.header.Authorization
          cors:
            origin: '*'
            headers:
              - Content-Type
              - X-Amz-Date
              - Authorization
              - X-Api-Key
              - X-Amz-Security-Token
              - X-Amz-User-Agent
            allowCredentials: false

  fun2:
    handler: src/f2.function2
    events:
      - http:
          method: post
          path: service/f2
          authorizer: # An AWS API Gateway custom authorizer function
            arn:  arn:aws:lambda:${opt:region, self:provider.region}:${self:provider.environment.ACCOUNT_ID}:function:${self:provider.environment.AUTHORIZER_FUNCTION} # Can be used instead of name to reference a function outside of service
            resultTtlInSeconds: 0
            identitySource: method.request.header.Authorization
          cors:
            origin: '*'
            headers:
              - Content-Type
              - X-Amz-Date
              - Authorization
              - X-Api-Key
              - X-Amz-Security-Token
              - X-Amz-User-Agent
            allowCredentials: false


  fun3:
    handler: src/f3.function3
    events:
      - http:
          method: post
          path: service/f3
          authorizer: # An AWS API Gateway custom authorizer function
            arn:  arn:aws:lambda:${opt:region, self:provider.region}:${self:provider.environment.ACCOUNT_ID}:function:${self:provider.environment.AUTHORIZER_FUNCTION} # Can be used instead of name to reference a function outside of service
            resultTtlInSeconds: 0
            identitySource: method.request.header.Authorization
          cors:
            origin: '*'
            headers:
              - Content-Type
              - X-Amz-Date
              - Authorization
              - X-Api-Key
              - X-Amz-Security-Token
              - X-Amz-User-Agent
            allowCredentials: false


plugins:
  - serverless-webpack
  - serverless-offline

package:
  individually: true

custom:
  serverless-offline:
    babelOptions:
      presets: ["es2017"]

  stage: "${opt:stage}"
```
-------------------------------------------------------------------------------------------------

5. Error Logs
-------------------------------------------------------------------------------------------------
```
Serverless: Uploading CloudFormation file to S3...
 Serverless: Uploading artifacts...
 Serverless: Uploading service f1.zip file to S3 (1.03 MB)...
 Serverless: Uploading service f2.zip file to S3 (1.03 MB)...
 Serverless: Uploading service f3.zip file to S3 (1.03 MB)...
 Serverless: Validating template...
 Serverless: Updating Stack...
 Serverless: Checking Stack update progress...
 CloudFormation - UPDATE_IN_PROGRESS - AWS::CloudFormation::Stack - stack-name
 CloudFormation - CREATE_IN_PROGRESS - AWS::ApiGateway::Authorizer - UserApiGatewayAuthorizer
 CloudFormation - CREATE_IN_PROGRESS - AWS::Lambda::Permission - UserLambdaPermissionApiGateway
 CloudFormation - CREATE_IN_PROGRESS - AWS::ApiGateway::Authorizer - UserApiGatewayAuthorizer
 CloudFormation - UPDATE_IN_PROGRESS - AWS::IAM::Role - IamRoleLambdaExecution
 CloudFormation - CREATE_FAILED - AWS::Lambda::Permission - UserLambdaPermissionApiGateway
 CloudFormation - CREATE_COMPLETE - AWS::ApiGateway::Authorizer - UserApiGatewayAuthorizer
 CloudFormation - UPDATE_FAILED - AWS::IAM::Role - IamRoleLambdaExecution
 CloudFormation - UPDATE_ROLLBACK_IN_PROGRESS - AWS::CloudFormation::Stack - stack-name
 CloudFormation - UPDATE_IN_PROGRESS - AWS::IAM::Role - IamRoleLambdaExecution
 CloudFormation - UPDATE_COMPLETE - AWS::IAM::Role - IamRoleLambdaExecution
 CloudFormation - UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS - AWS::CloudFormation::Stack - stack-name
 CloudFormation - DELETE_IN_PROGRESS - AWS::ApiGateway::Authorizer - UserApiGatewayAuthorizer
 CloudFormation - DELETE_COMPLETE - AWS::Lambda::Permission - UserLambdaPermissionApiGateway
 CloudFormation - DELETE_COMPLETE - AWS::ApiGateway::Authorizer - UserApiGatewayAuthorizer
 CloudFormation - UPDATE_ROLLBACK_COMPLETE - AWS::CloudFormation::Stack - stack-name
 Serverless: Operation failed!
 Serverless: View the full error output: https://ap-south-1.console.aws.amazon.com/cloudformation/home?region=ap-south-1#/stack/detail?stackId=arn%3Aaws%3Acloudformation%3Aap-south-1%3A236585061287%3Astack%2Fstack-name%2Fe176a640-4f0d-11ea-a0db-028f38a10716
  
   Serverless Error ---------------------------------------
  
   An error occurred: UserLambdaPermissionApiGateway - The final policy size (20691) is bigger than the limit (20480). (Service: AWSLambda; Status Code: 400; Error Code: PolicyLengthExceededException; Request ID: 4e1edf8f-6068-4557-84c1-93cb676c178d).
  
   Get Support --------------------------------------------
      Docs:          docs.serverless.com
      Bugs:          github.com/serverless/serverless/issues
      Issues:        forum.serverless.com
```
  
   Your Environment Information ---------------------------
      Operating System:          linux
      Node Version:              10.19.0
      Framework Version:         1.65.0
-------------------------------------------------------------------------------------------------

Please share you insights if i am doing anything wrong.

**Comment by medikoo** [comment]
@akshay-scalex thanks for report.

It looks that you were already close to limit, and recent security improvement (https://github.com/serverless/serverless/pull/7300) brought you above it.

At this point it's difficult for me to guide you how to recover from it. What would definitely help is reducing number of lambda resources or splitting given service into multiple services

**Comment by akshay-scalex** [comment]
thanks for the response @medikoo 

As of now I am having only 3 lambda function inside the serverless.yml and one authorizer is attached to it to secure the path. 

I have also noticed one more issue, that the same configurations are getting deployed in dev env as well as I noticed when you deploy the code after deleting the stack it works sometimes. But it is not allowing me to update the existing functions.
